PrivateHospitals/backend/PrivateHospitals/PrivateHospitals.Application/Services/Doctor/DoctorService.cs

```diff
@@ -0,0 +1,106 @@
+    private readonly IUserRepository _userRepository;
+
+    public DoctorService(IDoctorRepository doctorRepository, IValidator<RegisterDto> registerDtoValidator, IValidator<RegisterDoctorDto> registerDoctorDtoValidator, IUserRepository userRepository)
+    {
+        _doctorRepository = doctorRepository;
+        _registerDtoValidator = registerDtoValidator;
+        _registerDoctorDtoValidator = registerDoctorDtoValidator;
+        _userRepository = userRepository;
+    }
+
+
+    public async Task<ServiceResponse<bool>> RegisterDoctorAsync(RegisterDoctorDto doctorDto)
+    {   
+        //Register to Doctors
+        ValidationResult validationDoctorResult = await _registerDoctorDtoValidator.ValidateAsync(doctorDto);
+        
+        if (!validationDoctorResult.IsValid)
+        {
+            var errors = validationDoctorResult.Errors
+                .Select(error => error.ErrorMessage)
+                .ToList();
+
+            return ServiceResponse<bool>.ErrorResponse(errors);
+        }
+        
+        Core.Models.Doctor doctor = new Core.Models.Doctor()
+        {
+            FirstName = doctorDto.FirstName,
+            Email = doctorDto.Email,
+            LastName = doctorDto.LastName,
+            UserName = doctorDto.UserName,
+            Speciality = doctorDto.Speciality
+        };
+
+        var doctorResult = await _doctorRepository.AddDoctorAsync(doctor);
+        
+        if (doctorResult == false)
+        {
+            return ServiceResponse<bool>.ErrorResponse(new List<string>() { "Error creating doctor." });
+        }
+        
+        //Register To Users
+        RegisterDto registerDto = new RegisterDto()
+        {
+            FirstName = doctorDto.FirstName,
+            LastName = doctorDto.LastName,
+            Email = doctorDto.Email,
+            Username = doctorDto.UserName,
+            Password = doctorDto.Password
+        };
+        
+        ValidationResult validationUserResult = await _registerDtoValidator.ValidateAsync(registerDto);
+        if (!validationUserResult.IsValid)
+        {
+            var errors = validationUserResult.Errors.Select(e => e.ErrorMessage).ToList();
+
+            return ServiceResponse<bool>.ErrorResponse(errors);
+        }
+        
+        var user = new AppUser()
+        {
+            FirstName = registerDto.FirstName,
+            LastName = registerDto.LastName,
+            UserName = registerDto.Username,
+            Email = registerDto.Email
+        };
+        
+        var identityResult = await _userRepository.CreateUserAsync(user, registerDto.Password);
+        
+        if (!identityResult.Succeeded)
+        {
+            var errors = identityResult.Errors.Select(e => e.Description).ToList();
+            return ServiceResponse<bool>.ErrorResponse(errors);
+        }
+        
+        var roleResult =  await _userRepository.AddUserToRoleAsync(user, "Doctor");
```

Comment:
Shift `"Doctor"` role to Domain layer and create constants for it. This way u avoid typos and make your Domain layer more rich

Reply:
Fixed